### ISSUE TITLE
vuex server tables not compatible with vue-turbolinks

### ISSUE THREAD

**Issue by vesper8** [matfish2/vue-tables-2#339]
I know this is probably out of scope but since Turbolinks has been gaining a lot in popularity and I consider this package to be very popular as well, I wanted to point out that there's a problem when using server tables with vuex mode enabled and using turbolinks with the vue adapter

I opened an issue here https://github.com/jeffreyguenther/vue-turbolinks/issues/17

But was hoping you might have some idea why this is happening. It works fine when vuex mode is set to false.

**Comment by matfish2** [comment]
I need more details. What sort of problem? Do you get a console error?

**Comment by vesper8** [comment]
Yes this is the console error I get when using vuex mode and hitting the back button

```
TypeError: Cannot set property 'outerHTML' of undefined
    at Vue$3.destroyed (index.js:24)
    at callHook (vue.js:2872)
    at Vue$3.Vue.$destroy (vue.js:2682)
    at vuex.esm.js:567
    at Array.<anonymous> (vue.js:1798)
    at flushCallbacks (vue.js:1719)
    at <anonymous>
logError @ vue.js:1702
globalHandleError @ vue.js:1693
handleError @ vue.js:1682
callHook @ vue.js:2874
Vue.$destroy @ vue.js:2682
(anonymous) @ vuex.esm.js:567
(anonymous) @ vue.js:1798
flushCallbacks @ vue.js:1719
Promise resolved (async)
microTimerFunc @ vue.js:1767
nextTick @ vue.js:1811
queueWatcher @ vue.js:3018
update @ vue.js:3153
notify @ vue.js:686
del @ vue.js:1069
(anonymous) @ vuex.esm.js:482
_withCommit @ vuex.esm.js:495
unregisterModule @ vuex.esm.js:480
webpackJsonp../node_modules/vue-tables-2/compiled/state/register-module.js.module.exports @ register-module.js:21
webpackJsonp../node_modules/vue-tables-2/compiled/mixins/created.js.module.exports @ created.js:10
created @ v-server-table.js:65
callHook @ vue.js:2872
Vue._init @ vue.js:4504
VueComponent @ vue.js:4674
createComponentInstanceForVnode @ vue.js:4195
init @ vue.js:4012
createComponent @ vue.js:5463
createElm @ vue.js:5411
createChildren @ vue.js:5534
createElm @ vue.js:5439
createChildren @ vue.js:5534
createElm @ vue.js:5439
patch @ vue.js:5928
Vue._update @ vue.js:2617
updateComponent @ vue.js:2745
get @ vue.js:3086
Watcher @ vue.js:3075
mountComponent @ vue.js:2749
./node_modules/vue/dist/vue.js.Vue$3.$mount @ vue.js:8322
./node_modules/vue/dist/vue.js.Vue$3.$mount @ vue.js:10661
init @ vue.js:4018
createComponent @ vue.js:5463
createElm @ vue.js:5411
createChildren @ vue.js:5534
createElm @ vue.js:5439
createChildren @ vue.js:5534
createElm @ vue.js:5439
createChildren @ vue.js:5534
createElm @ vue.js:5439
patch @ vue.js:5928
Vue._update @ vue.js:2617
updateComponent @ vue.js:2745
get @ vue.js:3086
Watcher @ vue.js:3075
mountComponent @ vue.js:2749
./node_modules/vue/dist/vue.js.Vue$3.$mount @ vue.js:8322
./node_modules/vue/dist/vue.js.Vue$3.$mount @ vue.js:10661
init @ vue.js:4018
createComponent @ vue.js:5463
createElm @ vue.js:5411
createChildren @ vue.js:5534
createElm @ vue.js:5439
createChildren @ vue.js:5534
createElm @ vue.js:5439
patch @ vue.js:5967
Vue._update @ vue.js:2617
updateComponent @ vue.js:2745
get @ vue.js:3086
Watcher @ vue.js:3075
mountComponent @ vue.js:2749
./node_modules/vue/dist/vue.js.Vue$3.$mount @ vue.js:8322
./node_modules/vue/dist/vue.js.Vue$3.$mount @ vue.js:10661
Vue._init @ vue.js:4514
Vue$3 @ vue.js:4599
(anonymous) @ app.js:49
t.dispatch @ turbolinks.js:5
t.Controller.r.notifyApplicationAfterPageLoad @ turbolinks.js:5
t.Controller.r.visitCompleted @ turbolinks.js:6
t.Visit.r.complete @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
requestAnimationFrame (async)
t.Visit.r.render @ turbolinks.js:5
t.Visit.r.loadResponse @ turbolinks.js:5
t.BrowserAdapter.r.visitRequestCompleted @ turbolinks.js:5
t.Visit.r.requestCompletedWithResponse @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
t.HttpRequest.r.endRequest @ turbolinks.js:5
t.HttpRequest.r.requestLoaded @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
XMLHttpRequest.send (async)
(anonymous) @ VM57233:1
t.HttpRequest.r.send @ turbolinks.js:5
t.Visit.r.issueRequest @ turbolinks.js:5
t.BrowserAdapter.r.visitStarted @ turbolinks.js:5
t.Visit.r.start @ turbolinks.js:5
t.Controller.r.startVisit @ turbolinks.js:5
t.Controller.r.startVisitToLocationWithAction @ turbolinks.js:5
t.BrowserAdapter.r.visitProposedToLocationWithAction @ turbolinks.js:5
t.Controller.r.visit @ turbolinks.js:5
t.Controller.r.clickBubbled @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
vue.js:580 [Vue warn]: Error in destroyed hook: "TypeError: Cannot set property 'outerHTML' of undefined"

(found in <Root>)
```

**Comment by excid3** [comment]
Hmmm, can you provide a small example repo that reproduces the problem?

**Comment by vesper8** [comment]
If it will help you help me then it's my pleasure to have taken the time to make an example repo :)

Can you tell I really want this solved.. it's really causing me problems..

Here's the example repo that does demonstrate the problem with very minimal setup

https://github.com/vesper8/debugging-vuex-turbolinks

I created two distincts app.js because the very existence of another table that implements Vuex, even if that component is not present on the page.. causes the error

So I created one app-with-vuex.js and one app-without-vuex.js and they are completely separate

Here's the files I customized
https://github.com/vesper8/debugging-vuex-turbolinks/tree/master/resources/assets/js
https://github.com/vesper8/debugging-vuex-turbolinks/tree/master/resources/views

And I installed two packages, this one, and the vue-turbolinks which is required

```
    "dependencies": {
        "vue-tables-2": "^0.6.97",
        "vue-turbolinks": "^2.0.0"
    }
```

Just clone the repo, run composer install and you should be good to test

If you visit 
https://debugging-vuex-turbolinks.dev/without-vuex

Then you'll see you can click on the links and the console stays clean. You can either click on the fodder link and back, or revisit the same page with the table component

Doing the same thing with the vuex table:
https://debugging-vuex-turbolinks.dev/with-vuex

Immediately causes an error.. assumedly during the teardown process

Hope you guys can help @excid3 and @matfish2

**Comment by matfish2** [comment]
I've isolated the source of the problem to calls to `regsiterModule` and `unregisterModule` in `state/register-module.js` and `mixins/before-destroy.js`, respectively.

I think the problem is that when these methods are called a `destroy` hook is called, but by the time it runs the page no longer has the root element, because turbolinks has already replaced the content area.

To test my assumption see if you can delay the execution of turbolinks click event a little bit. If I am right, the error should be gone.

**Comment by vesper8** [comment]
Hi @matfish2 !

First of all I wanna say congrats on 1.0!

Your package continues to be at the center of many of my projects and I'm so happy you're intent on continuing to polish it and add functionalities!

I'm also so glad to see that you're actively working on fixing this particular problem!

Now here's my findings.

I installed the latest version (1.0.10) and the first thing I noticed, is that I am now getting the outerHtml error whether I use vuex or not! :(

I then tried adding preserveState: true to my tableOptions and I can't tell if that's changing anything but it unfortunately does not make the error go away.

Lastly I tried to implement your advice on adding a delay before allowing Turbolinks to do its thing.

Looking at the available list of events: document.addEventListener("turbolinks:click", function (event) {
    console.log('clicking...' + event.data.url);
    sleep(2000);
});
document.addEventListener("turbolinks:visit", function (event) {
    console.log('visiting...' + event.data.url);
    sleep(2000);
});

I tried adding a pause on both the click and visit event, and unfortunately while this pause does work, it doesn't help with the problem

```
document.addEventListener("turbolinks:click", function (event) {
    console.log('clicking...' + event.data.url);
    sleep(2000);
});
document.addEventListener("turbolinks:visit", function (event) {
    console.log('visiting...' + event.data.url);
    sleep(2000);
});

function sleep(miliseconds) {
    const currentTime = new Date().getTime();

    while (currentTime + miliseconds >= new Date().getTime()) {}
}
```

I suppose you must have already read this. Perhaps having another look-over will provide some insight as to what is the best way to circumvent this problem

https://github.com/turbolinks/turbolinks/wiki/VueJs-and-Turbolinks
 
As I mentioned above, I already am using the vue-turbolinks adapter mentioned in the above link 

I feel you're very close to solving this!

**Comment by vesper8** [comment]
I tried adding delays on all possible events

```
document.addEventListener("turbolinks:load", function (event) {
    console.log('load');
    sleep(1000);
});
document.addEventListener("turbolinks:click", function (event) {
    console.log('click');
    sleep(1000);
});
document.addEventListener("turbolinks:before-visit", function (event) {
    console.log('before-visit');
    sleep(1000);
});
document.addEventListener("turbolinks:visit", function (event) {
    console.log('visit');
    sleep(1000);
});
document.addEventListener("turbolinks:request-start", function (event) {
    console.log('request-start');
    sleep(1000);
});
document.addEventListener("turbolinks:request-end", function (event) {
    console.log('request-end');
    sleep(1000);
});
document.addEventListener("turbolinks:before-cache", function (event) {
    console.log('before-cache');
    sleep(1000);
});
document.addEventListener("turbolinks:before-render fires", function (event) {
    console.log('before-render fires');
    sleep(1000);
});
document.addEventListener("turbolinks:render_fires", function (event) {
    console.log('render_fires');
    sleep(1000);
});
document.addEventListener("turbolinks:load fires", function (event) {
    console.log('load fires');
    sleep(1000);
});

function sleep(miliseconds) {
    const currentTime = new Date().getTime();

    while (currentTime + miliseconds >= new Date().getTime()) {}
}
```

Didn't help unfortunately. But it may give you insight on what's going on.

The error always occurs between the turbolinks:visit event and the turbolinks:request-end event

There are two errors as well. Here they are:

```
[Vue warn]: Error in destroyed hook: "TypeError: Cannot set property 'outerHTML' of undefined"

(found in <Root>)
warn @ entry-client.js?id=b67608ed72091e4bbb81:69524
logError @ entry-client.js?id=b67608ed72091e4bbb81:70656
globalHandleError @ entry-client.js?id=b67608ed72091e4bbb81:70651
handleError @ entry-client.js?id=b67608ed72091e4bbb81:70640
callHook @ entry-client.js?id=b67608ed72091e4bbb81:71835
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71640
webpackJsonp../node_modules/vue-tables-2/compiled/mixins/before-destroy.js.module.exports @ 16.js:19702
callHook @ entry-client.js?id=b67608ed72091e4bbb81:71833
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71615
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
teardown @ entry-client.js?id=b67608ed72091e4bbb81:68910
t.dispatch @ turbolinks.js:5
t.Controller.r.notifyApplicationAfterVisitingLocation @ turbolinks.js:5
t.Controller.r.startVisit @ turbolinks.js:5
t.Controller.r.startVisitToLocationWithAction @ turbolinks.js:5
t.BrowserAdapter.r.visitProposedToLocationWithAction @ turbolinks.js:5
t.Controller.r.visit @ turbolinks.js:5
t.Controller.r.clickBubbled @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
entry-client.js?id=b67608ed72091e4bbb81:70660 TypeError: Cannot set property 'outerHTML' of undefined
    at Vue$3.destroyed (entry-client.js?id=b67608ed72091e4bbb81:68931)
    at callHook (entry-client.js?id=b67608ed72091e4bbb81:71833)
    at Vue$3.Vue.$destroy (entry-client.js?id=b67608ed72091e4bbb81:71640)
    at VueComponent.webpackJsonp../node_modules/vue-tables-2/compiled/mixins/before-destroy.js.module.exports (16.js:19702)
    at callHook (entry-client.js?id=b67608ed72091e4bbb81:71833)
    at VueComponent.Vue.$destroy (entry-client.js?id=b67608ed72091e4bbb81:71615)
    at destroy (entry-client.js?id=b67608ed72091e4bbb81:73022)
    at invokeDestroyHook (entry-client.js?id=b67608ed72091e4bbb81:74545)
    at invokeDestroyHook (entry-client.js?id=b67608ed72091e4bbb81:74550)
    at invokeDestroyHook (entry-client.js?id=b67608ed72091e4bbb81:74550)
logError @ entry-client.js?id=b67608ed72091e4bbb81:70660
globalHandleError @ entry-client.js?id=b67608ed72091e4bbb81:70651
handleError @ entry-client.js?id=b67608ed72091e4bbb81:70640
callHook @ entry-client.js?id=b67608ed72091e4bbb81:71835
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71640
webpackJsonp../node_modules/vue-tables-2/compiled/mixins/before-destroy.js.module.exports @ 16.js:19702
callHook @ entry-client.js?id=b67608ed72091e4bbb81:71833
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71615
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
teardown @ entry-client.js?id=b67608ed72091e4bbb81:68910
t.dispatch @ turbolinks.js:5
t.Controller.r.notifyApplicationAfterVisitingLocation @ turbolinks.js:5
t.Controller.r.startVisit @ turbolinks.js:5
t.Controller.r.startVisitToLocationWithAction @ turbolinks.js:5
t.BrowserAdapter.r.visitProposedToLocationWithAction @ turbolinks.js:5
t.Controller.r.visit @ turbolinks.js:5
t.Controller.r.clickBubbled @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
entry-client.js?id=b67608ed72091e4bbb81:69524 [Vue warn]: Error in destroyed hook: "TypeError: Cannot set property 'outerHTML' of undefined"

(found in <Root>)
warn @ entry-client.js?id=b67608ed72091e4bbb81:69524
logError @ entry-client.js?id=b67608ed72091e4bbb81:70656
globalHandleError @ entry-client.js?id=b67608ed72091e4bbb81:70651
handleError @ entry-client.js?id=b67608ed72091e4bbb81:70640
callHook @ entry-client.js?id=b67608ed72091e4bbb81:71835
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71640
beforeDestroy @ 16.js:21815
callHook @ entry-client.js?id=b67608ed72091e4bbb81:71833
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71615
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
destroy @ entry-client.js?id=b67608ed72091e4bbb81:73022
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74545
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
invokeDestroyHook @ entry-client.js?id=b67608ed72091e4bbb81:74550
patch @ entry-client.js?id=b67608ed72091e4bbb81:74883
Vue.$destroy @ entry-client.js?id=b67608ed72091e4bbb81:71638
teardown @ entry-client.js?id=b67608ed72091e4bbb81:68910
t.dispatch @ turbolinks.js:5
t.Controller.r.notifyApplicationAfterVisitingLocation @ turbolinks.js:5
t.Controller.r.startVisit @ turbolinks.js:5
t.Controller.r.startVisitToLocationWithAction @ turbolinks.js:5
t.BrowserAdapter.r.visitProposedToLocationWithAction @ turbolinks.js:5
t.Controller.r.visit @ turbolinks.js:5
t.Controller.r.clickBubbled @ turbolinks.js:5
(anonymous) @ turbolinks.js:5
entry-client.js?id=b67608ed72091e4bbb81:70660 TypeError: Cannot set property 'outerHTML' of undefined
    at Vue$3.destroyed (entry-client.js?id=b67608ed72091e4bbb81:68931)
    at callHook (entry-client.js?id=b67608ed72091e4bbb81:71833)
    at Vue$3.Vue.$destroy (entry-client.js?id=b67608ed72091e4bbb81:71640)
    at VueComponent.beforeDestroy (16.js:21815)
    at callHook (entry-client.js?id=b67608ed72091e4bbb81:71833)
    at VueComponent.Vue.$destroy (entry-client.js?id=b67608ed72091e4bbb81:71615)
    at destroy (entry-client.js?id=b67608ed72091e4bbb81:73022)
    at invokeDestroyHook (entry-client.js?id=b67608ed72091e4bbb81:74545)
    at invokeDestroyHook (entry-client.js?id=b67608ed72091e4bbb81:74550)
    at VueComponent.patch [as __patch__] (entry-client.js?id=b67608ed72091e4bbb81:74883)
```

**Comment by vesper8** [comment]
w00t w00t!! man this has been giving me so much grief! but I think it's finally working now

this commit made the error go away https://github.com/jeffreyguenther/vue-turbolinks/commit/7f040a1c5f90c9c8078d05dd6a74b6b473a9e01a

and then I spent hours debugging why the pagination events were still being duplicated.. then I realized I had set preserveState: true on your suggestion.. and that was actually the culprit!

I set it back to false.. and no more duplication! no more red errors!! my app is starting to run pretty damn smoothly!! yay
